docker/Dockerfile

```diff
@@ -0,0 +1,58 @@
+FROM gcc:11 as build
+
+WORKDIR /build
+
+RUN --mount=type=cache,id=build-apt-cache,target=/var/cache/apt \
+    echo "Install build dependencies" && \
+    apt update && \
+    apt remove -y libmysqlcppconn7v5 libmysqlcppconn-dev && \
+    apt install cmake zlib1g zlib1g-dev unzip -yqq --no-install-recommends && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY dAuthServer/ /build/dAuthServer
+COPY dChatServer/ /build/dChatServer
+COPY dCommon/ /build/dCommon
+COPY dChatFilter/ /build/dChatFilter
+COPY dDatabase/ /build/dDatabase
+COPY dGame/ /build/dGame
+COPY dMasterServer/ /build/dMasterServer
+COPY dNet/ /build/dNet
+COPY dPhysics/ /build/dPhysics
+COPY dScripts/ /build/dScripts
+COPY dWorldServer/ /build/dWorldServer
+COPY dZoneManager/ /build/dZoneManager
+COPY migrations/ /build/migrations
+COPY resources/ /build/resources
+COPY thirdparty/ /build/thirdparty
+COPY vanity /build/vanity
+COPY .clang-* CMake* LICENSE /build/
+
+ARG BUILD_THREADS=1
+ARG BUILD_VERSION=171022
+
+RUN echo "Build server" && \
+    mkdir -p build && \
+    cd build && \
+    sed -i -e "s/171022/${BUILD_VERSION}/g" ../CMakeVariables.txt && \
+    cmake .. && \
+    make -j $BUILD_THREADS
+
+RUN unzip /build/resources/navmeshes.zip -d /build/build/res/maps
+
+FROM gcc:11 as runtime
```

Comment:
+1 for multistage builds, nice!

Reply:
What is the rational for using `gcc:11` as the runtime?
I think the `gcc` container is based on Debian. Wouldn't it save space to use a base container that does not carry a full compiler?